sympy/matrices/expressions/matexpr.py

```diff
@@ -1026,6 +1035,44 @@ def _entry(self, i, j, **kwargs):
         return S.One
 
 
+class SingleEntryMatrix(MatrixExpr):
+    """A matrix which is zero everywhere except the entry (i, j) which
+    is 1.
```

Comment:
Would it make sense to store the value on the object rather than having it always be 1?